web/Controllers/PropertyController.cs

```diff
@@ -57,36 +58,46 @@ public async Task<IActionResult> AddNewProperty_Continue([FromRoute] int step, [
 
         // Get the property we're currently adding
         var property = _propertyService.GetIncompleteProperty(landlordId);
-        if (property == null)
+        if (newPropertyModel.Address.Postcode != null)
         {
-            if (step == 1)
+            newPropertyModel.Address.Postcode =
+                Regex.Replace(newPropertyModel.Address.Postcode, @"^(\S+?)\s*?(\d\w\w)$", "$1 $2");
+            newPropertyModel.Address.Postcode = newPropertyModel.Address.Postcode.ToUpper();
+        }
+
+        if (step == 1)
+        {
+            if (newPropertyModel.Address.AddressLine1 == null || newPropertyModel.Address.Postcode == null)
+            {
+                // Address line 1 and postcode are the minimum information we need to create a new record
+                return View("AddNewProperty",
+                    new AddNewPropertyViewModel { Step = step, Property = newPropertyModel });
+            }
+
+            PropertyViewModel resultModel = await _azureMapsApiService.AutofillAddress(newPropertyModel);
+            newPropertyModel = resultModel;

```

Comment:
The method mutates it anyway, so just this should work

```suggestion
            await _azureMapsApiService.AutofillAddress(newPropertyModel);
```

Might be worth changing `AutofillAddress` to just return a `Task` to make that clear